src/components/layer-controller/LayerControllerSubscriber.js

```diff
@@ -6,29 +6,52 @@ import RasterLayerController from './RasterLayerController';
 import VectorLayerController from './VectorLayerController';
 import ImageAddButton from './ImageAddButton';
 import { useReady } from '../hooks';
-import { useRasterData } from '../data-hooks';
+import { useCellsData, useMoleculesData, useRasterData } from '../data-hooks';
 import { useCoordination, useLoaders } from '../../app/state/hooks';
 import { COMPONENT_COORDINATION_TYPES } from '../../app/state/coordination';
 import { initializeLayerChannels } from '../spatial/utils';
 import { DEFAULT_RASTER_LAYER_PROPS } from '../spatial/constants';
 
 const LAYER_CONTROLLER_DATA_TYPES = ['raster'];
 
+/**
+ * A subscriber component for the spatial layer controller.
+ * @param {object} props
+ * @param {string} props.theme The current theme name.
+ * @param {object} props.coordinationScopes The mapping from coordination types to coordination
+ * scopes.
+ * @param {function} props.removeGridComponent The callback function to pass to TitleInfo,
+ * to call when the component has been removed from the grid.
+ * @param {boolean} props.initializeSpatialRasterLayers Should the coordination
+ * value be automatically initialized based on the data?
+ * @param {boolean} props.initializeSpatialCellsLayer Should the coordination
+ * value be automatically initialized based on the data?
+ * @param {boolean} props.initializeSpatialMoleculesLayer Should the coordination
+ * value be automatically initialized based on the data?
```

Comment:
Can you give a bit of background to these props?  I see how they affect the change, for example in the `wang-2019` dataset, but it feels like it's masking an issue, although I might have that wrong.  If someone passes in settings in a view config, that should be respected regardless of what some parameter passed into the subscriber says, no?  I could see this causing issues in the future if someone creates a setting, saves the view config (like in the portal), but this feature is not changed to `false` (Like in `wang-2019`, which has custom moleceules settings).  Then their saved state will be overriden, no?

Reply:
Yes this is a good point. Also having these `initialize` props will probably make writing the `upgrade()` functions to upgrade view configs pretty tricky in the future if the names of the coordination types change (and they likely will change a bit like they did here for the spatial layers).

So what you are saying is that we should remove all of these `initialize` props and instead, if someone "does not want to initialize" a coordination type, they should set its value to an "empty value" rather than a "null value". For instance, to "not initialize" the raster layers automatically, the user would set `spatialRasterLayers` to the empty array `[]` in the view config? Whereas if they had set `spatialRasterLayers` to `null` the value would have been set automatically?

I guess I am not sure how this would work for something like a string value, where I wouldnt want to be using the empty string as a special value. So maybe `null` should be the "empty value" and auto-initializations should require a magic string like "__auto__"?

Reply:
Just trying to understand, but what is the use case for not initializing values at all?  My sense is that that would break things - how would we know where to put the sliders, or what the radius of the cells is?

Reply:
It seems like the old way only had two options - pass in values or they will be initialized for you, but I might have that wrong.

Reply:
You are right. We currently have this option to prevent the cell sets from being initialized https://github.com/vitessce/vitessce/blob/master/src/components/sets/CellSetsManagerSubscriber.js#L75 and I was trying to generalize this, but I can't think of a real use case for this, so I think we should remove these props and simplify things.

I think we need to remove these [`coordinationPreferences` values](https://github.com/vitessce/vitessce/blob/keller-mark/coordination-init/src/components/data-hooks.js#L633) and instead pass in the current coordination values (the [first element of the array returned by `useCoordination`](https://github.com/vitessce/vitessce/blob/keller-mark/coordination-init/src/components/spatial/SpatialSubscriber.js#L79-L96) in each component, so that we do not initialize something if it already has a value

Reply:
Ok thanks (multiple commits are no big deal though). Sorry this PR had so many issues.  I think all of these tweaks have/will improve it a lot

Reply:
No worries, the structure was the important part and you definitely nailed that!

Reply:
@keller-mark Just pushed a change.  Basically, I pass in the values themselves instead of the `initialize` boolean, and then check them against the initial value in the hook.  If they match the initial value, and there is a value coming from the loader, the loader initializes.  Otherwise, use the "initialized value"

Reply:
Can't request your review but it should all be in the one commit.

Reply:
Seems to work with all the examples, including OME-Zarr